File: Toy_Language_Interpreter/Expressions/ConstantExpression.java
package Model.Expressions;

import ADTs.IDictionary;
import Heap.IHeap;

public class ConstantExpression extends IExpression {
    private int number;

    public ConstantExpression(int num) {
        this.number = num;
    }

    public int eval(IDictionary<String, Integer> table, IHeap heap) {
        return this.number;
    }

    public String toString() {
        return Integer.toString(this.number); // sau return String.valueOf(this.number);
    }
}

File: Toy_Language_Interpreter_C/Expressions/ConstantExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class ConstantExpression : IExpression
    {
        private int number;

        public ConstantExpression(int num)
        {
            this.number = num;
        }

        public override int Eval(IDictionary<string, int> table)
        {
            return this.number;
        }

        public override string ToString()
        {
            return this.number.ToString();
        }
    }
}
